src/core/dbus-service.c

```
@@ -532,6 +562,43 @@ static int bus_service_set_transient_property(
         if (streq(name, "StandardErrorFileDescriptor"))
                 return bus_set_transient_std_fd(u, name, &s->stderr_fd, &s->exec_context.stdio_as_fds, message, flags, error);
 
+        if (streq(name, "OpenFile")) {
+                char *path = NULL;
+                char *fdname = NULL;
+                uint64_t options;
+
+                r = sd_bus_message_enter_container(message, 'a', "(sst)");
+                if (r < 0)
+                        return r;
+
+                while ((r = sd_bus_message_read(message, "(sst)", &path, &fdname, &options)) > 0) {
+                        if (UNIT_WRITE_FLAGS_NOOP(flags)) continue;
+                        OpenFile *of = new(OpenFile, 1);
+                        if (!of)
+                                return -ENOMEM;
+
+                        *of = (OpenFile) {
+                                .path = strdup(path),
+                                .fdname = strdup(fdname),
+                                .flags = options,
+                        };
+                        if (!of->path || !of->fdname)
+                                return -ENOMEM;
+
+                        if (!path_is_absolute(of->path))
+                                return -EINVAL;
+
```

We need a check here to verify that only defined flags were passed by the user. Otherwise, when we introduce a new flag, it could be turned on by accident for users if we silently ignore any undefined flags. Grep for `_PORTABLE_MASK_PUBLIC` to see how to do this. You'll need to introduce a similar constant for the open file flags enum